int/tests/conftest.py

```diff
@@ -0,0 +1,166 @@
+            url = r.json()["invitation_url"]
+            if not url:
+                await asyncio.sleep(1)
+        return json.loads(urlsafe_b64decode(url.split("c_i=")[1]))
+
+
+async def get_mediator_invite(external_mediator: Client) -> str:
+    invitation = await create_invitation.asyncio(
+        client=external_mediator, json_body=CreateInvitationRequest()
+    )
+    if not invitation:
+        raise RuntimeError("Failed to retrieve invitation from mediator")
+    assert not isinstance(invitation.invitation_url, Unset)
+    return invitation.invitation_url
+
+
+async def proxy_receive_mediator_invite(external_mediator: Client, invite: str):
+    async with AsyncClient() as client:
+        r = await client.post(
+            f"{PROXY}/receive_mediator_invitation", json={"invitation_url": invite}
+        )
+        assert not r.is_error
+
+    # Get mediator mediation requests
+    async def _retrieve():
+        requests = await get_mediation_requests.asyncio(
+            client=external_mediator, state=GetMediationRequestsState.GRANTED
+        )
+        assert requests
+        return requests
+
+    await poll_until_condition(
+        lambda reqs: bool(not isinstance(reqs.results, Unset) and reqs.results),
+        _retrieve,
+    )
+
+
+async def get_proxy_invite() -> dict:
+    async with AsyncClient() as client:
+        url = None
+        while url is None:
+            r = await client.get(f"{PROXY}/retrieve_agent_invitation")
+            url = r.json()["invitation_url"]
+            if not url:
+                await asyncio.sleep(1)
+        return json.loads(urlsafe_b64decode(url.split("c_i=")[1]))
+
+
+async def agent_receive_invitation(agent_bob: Client, invite: dict) -> ConnRecord:
+    conn_record = await receive_invitation.asyncio(
+        client=agent_bob, json_body=ReceiveInvitationRequest.from_dict(invite)
+    )
+    if not conn_record:
+        raise RuntimeError("Failed to receive invitation on agent")
+
+    async def _retrieve(connection_id: str) -> ConnRecord:
+        retrieved = await get_connection.asyncio(connection_id, client=agent_bob)
+        assert retrieved
+        return retrieved
+
+    await record_state("active", partial(_retrieve, conn_record.connection_id))
+    return conn_record
+
+
+async def agent_request_mediation_from_proxy(agent_bob: Client, conn_id: str):
+    mediation_record = await post_mediation_request_conn_id.asyncio(
+        conn_id=conn_id, client=agent_bob, json_body=MediationCreateRequest()
+    )
+    if not mediation_record:
+        raise RuntimeError(f"Failed to request mediation from {conn_id}")
+
+    while mediation_record.state != "granted":
+        await asyncio.sleep(1)
+        assert isinstance(mediation_record.mediation_id, str)
+        mediation_record = await get_mediation_requests_mediation_id.asyncio(
+            mediation_record.mediation_id, client=agent_bob
+        )
+        assert mediation_record
```

Comment:
This loop also needs replacement with one of our helpers to make sure it doesn't loop infinitely. This will look similar to the `record_state` check used in the `agent_receive_invitation` function.